store/store_download.go

```diff
@@ -343,15 +453,26 @@ func downloadImpl(ctx context.Context, name, sha3_384, downloadURL string, user
 		}
 		dlSize = float64(resp.ContentLength)
 		pbar.Start(name, dlSize)
-		mw := io.MultiWriter(w, h, pbar)
+		mw := io.MultiWriter(w, h, pbar, tc)
 		var limiter io.Reader
 		limiter = resp.Body
 		if limit := dlOpts.RateLimit; limit > 0 {
 			bucket := ratelimit.NewBucketWithRate(float64(limit), 2*limit)
 			limiter = ratelimitReader(resp.Body, bucket)
 		}
+
+		finish := tc.Monitor()
```

Comment:
(nitpick) I would be more explicit here and use "finished" or even "stopMonitorCh"